### DISCUSSION TITLE
[4.x] Workflow Notifications non admins

### DISCUSSION THREAD

**Discussion by brianteeman** [joomla/joomla-cms#47077] [2. Cold Features]
### Steps to reproduce the issue

- Steps to reproduce the issue
- Enable workflows in the content options with Basic as the default
- Edit all the Transitions so they have notifications enabled and set the Usergroups to Publisher
- Ensure that you have users in that group
- Create a new article and save & close
- Edit an article and change the stage

### Expected result
Notification email sent 
![image](https://user-images.githubusercontent.com/1296369/166883093-18da6050-8e93-4425-8dbd-937e283e763e.png)

### Actual result
Message says it has been sent but nothing is received
![image](https://user-images.githubusercontent.com/1296369/166883142-44c67748-a245-455d-a97f-7310b363bf58.png)

**Comment by brianteeman** [comment]
The problem is coming from 
`				if ($receiver->authorise('core.manage', 'com_message'))`

This restricts the receivers to those with admin access and com_message access.

That's fine if it is the intended behaviour but the implication of the documentation is that it is not the intended behaviour https://docs.joomla.org/Publishing_Workflow/Scenarios/en#Setting_up_Notifications

It would probably be better to send the mail using getMailer instead of com_messages and thats a very easy change to make

**Comment by roland-d** [comment]
I am working on a PR for exactly that. More details will follow later

**Comment by brianteeman** [comment]
ok I will undo what I have been working on

**Comment by brianteeman** [comment]
For reference I was following the pattern used for actionlogs as that would make it indeoendent of com_messages and allow for com_,mail_templates support

**Comment by roland-d** [comment]
@brianteeman I like the idea of integrating it into the mail templates. Just I was wondering if we should keep the com_messages there as well. If we don´t use com_messages, what is the use of com_messages but that may be a complete different discussion :smile:

**Comment by brianteeman** [comment]
If you use com_messages for this then you are restricted to users who have admin access.

com_messages is still for used for many things
